contracts/interfaces/IStakingPool.sol

```diff
@@ -6,46 +6,93 @@ import "@openzeppelin/contracts-v4/token/ERC721/IERC721.sol";
 
 interface IStakingPool is IERC721 {
 
-  struct Weight {
+  /* structs for storage */
+
+  // stakers are grouped based on the timelock expiration
+  // group index is calculated based on the expiration date
+  // the initial proposal is to have 4 groups per year (1 group per quarter)
+  struct StakeGroup {
+    uint stakeShares;
+    uint rewardsShares;
+    uint groupSharesSupply;
+    // TODO: consider extracting the following fields to a separate struct
+    uint accRewardPerStakeShareAtExpiration;
+    uint expiredStakeAmount;
+  }
+
+  struct Product {
+    uint weight;
+    uint allocatedStake;
+    uint lastBucket;
+  }
+
+  struct PoolBucket {
+    uint rewardPerSecondCut;
+  }
+
+  struct ProductBucket {
+    uint allocationCut;
+  }
+
+  /* structs for io */
+
+  struct WithdrawParams {
+    uint positionId;
+    uint[] groupIds;
+    uint flags;
+  }
+
+  struct ProductParams {
```

Comment:
Move structs outside of the interface so they can be used by other contracts by importing this file and not having to specify the name of the interface for each of them.

Reply:
Done